src/Utils/Helpers.ts

```diff
@@ -88,4 +89,15 @@ const isInComponents = (path: string): boolean => {
 const isInChannels = (path: string): boolean => {
   return path.startsWith('channels.');
 };
-export { compareComponents, isEqual, isInComponents, isInChannels };
+const toJS = (asyncapiYAMLorJSON: any): any => {
+  if (asyncapiYAMLorJSON.constructor && asyncapiYAMLorJSON.constructor.name === 'Object') {
+    return JSON.parse(JSON.stringify(asyncapiYAMLorJSON));
+  } 
+  if (typeof asyncapiYAMLorJSON === 'string') {
+    if (asyncapiYAMLorJSON.trimLeft().startsWith('{')) {
+      return JSON.parse(asyncapiYAMLorJSON);
+    } 
+    return YAML.parse(asyncapiYAMLorJSON);
+  }
```

Comment:
`trimLeft` is supported in major browsers, but better will be if you will make it with `trim()` :)

```suggestion
  if (typeof asyncapiYAMLorJSON === 'string') {
    if (asyncapiYAMLorJSON.trim().startsWith('{')) {
      return JSON.parse(asyncapiYAMLorJSON);
    } 
    return YAML.parse(asyncapiYAMLorJSON);
  }
```

Also as I remember `YAML.parse` can parse JSON and YAML so you can only make this:

```suggestion
  if (typeof asyncapiYAMLorJSON === 'string') {
    return YAML.parse(asyncapiYAMLorJSON);
  }
```

Reply:
Sorry for this suggestion! In other our repos we are using the [`js-yaml`](https://www.npmjs.com/package/js-yaml) library which has function `load` which can parse both formats, JSON and YAML, so please change the dependency from `yaml` package to the `js-yaml` and change lines to:

```js
  if (typeof asyncapiYAMLorJSON === 'string') {
    return YAML.load(asyncapiYAMLorJSON);
  }
```

Reply:
@KhudaDad414 Did you update code with my suggestion?

Reply:
Ok I see it, sorry :) I saw on previous revision.